src/orca/services/nextflowtower/models.py

```diff
@@ -0,0 +1,225 @@
+        """
+        return cls(response["id"], response["userName"], response["email"])
+
+
+@dataclass(kw_only=False)
+class Organization:
+    """Nextflow Tower organization."""
+
+    id: int
+    name: str
+
+    @classmethod
+    def from_response(cls, response: dict[str, Any]) -> Self:
+        """Create organization from API JSON response.
+
+        Returns:
+            Organization instance.
+        """
+        return cls(response["orgId"], response["orgName"])
+
+
+@dataclass(kw_only=False)
+class Workspace:
+    """Nextflow Tower workspace."""
+
+    id: int
+    name: str
+    org: Organization
+
+    @property
+    def full_name(self) -> str:
+        """Fully-qualified workspace name (with organization name)."""
+        return f"{self.org.name}/{self.name}".lower()
+
+    @classmethod
+    def from_response(cls, response: dict[str, Any]) -> Self:
+        """Create workspace from API JSON response.
+
+        Returns:
+            Workspace instance.
+        """
+        org = Organization.from_response(response)
+        return cls(response["workspaceId"], response["workspaceName"], org)
+
+
+@dataclass(kw_only=False)
+class LaunchInfo:
+    """Nextflow Tower workflow launch specification."""
+
+    compute_env_id: str
+    pipeline: str
+    work_dir: str
+    revision: Optional[str] = None
+    params: Optional[dict] = None
+    nextflow_config: Optional[str] = None
+    run_name: Optional[str] = None
+    pre_run_script: Optional[str] = None
+    profiles: list[str] = field(default_factory=list)
+    user_secrets: list[str] = field(default_factory=list)
+    workspace_secrets: list[str] = field(default_factory=list)
+    label_ids: list[int] = field(default_factory=list)
+
+    @staticmethod
+    def dedup(items: Collection[str]) -> list[str]:
+        """Deduplicate items in a collection.
+
+        Args:
+            items: Collection of items.
+
+        Returns:
+            Deduplicated collection or None.
+        """
+        return list(set(items))
+
+    @staticmethod
+    def get_current_timestamp():
+        """Generate current timestamp using Tower's UTC format.
+
+        Returns:
+            Current timestamp.
+        """
+        return datetime.now().isoformat()[:-3] + "Z"
+
+    def fill_in(self, attr: str, value: Any):
+        """Fill in any missing values.
+
+        Args:
+            attr: Attribute name.
+            value: Attribute value.
+        """
+        if not getattr(self, attr, None):
+            setattr(self, attr, value)
+
+    def to_dict(self) -> dict[str, Any]:
+        """Generate JSON representation of a launch specification.
+
+        Returns:
+            JSON representation.
+        """
+        output = {
+            "launch": {
+                "computeEnvId": self.compute_env_id,
+                "configProfiles": self.dedup(self.profiles),
+                "configText": self.nextflow_config,
+                "dateCreated": self.get_current_timestamp(),
```

Comment:
I'm emulating the behavior that I'm seeing with the Tower web client. I'll try to launch a workflow without this field. If I can, then I'll drop this because it's simpler. IMHO, this feels like something that the backend should handle.

Reply:
I was able to run the `test_that_a_workflow_can_be_launched` integration test while setting this field to None, so I'm going to remove the timestamp code.